### PR TITLE
RUMM-3201 Add Roku documentation

### PR DESCRIPTION
**Pull request by xgouchet**
### What does this PR do?

Add the documentation for the RUM Roku sdk.

### Motivation

Document our new product. 

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/real_user_monitoring/roku/_index.md [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+{{< /site-region >}}
+
+{{< site-region region="us,us3,us5,eu,ap1" >}}
+<div class="alert alert-info">RUM for Roku is in beta.</div>
+
+## Overview
+
+Datadog Real User Monitoring (RUM) enables you to visualize and analyze the real-time performance and user journeys of your channel's individual users.
+
+The Datadog Roku SDK supports Brightscript channels for Roku OS 10 and higher.
```
Comment (cswatt) [inline content/en/real_user_monitoring/roku/_index.md:26]:
```suggestion
The Datadog Roku SDK supports BrightScript channels for Roku OS 10 and higher.
```

File: content/en/real_user_monitoring/roku/_index.md [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+
+## Setup
+
+1. Declare the SDK as a dependency.
+2. Specify application details in the UI.
```
Comment (cswatt) [inline content/en/real_user_monitoring/roku/_index.md:31]:
```suggestion
2. Specify application details in Datadog.
```

File: content/en/real_user_monitoring/roku/_index.md [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+```shell
+ropm install datadog-roku
+```
+
+### Setup manually
+
+If your project does not use `ROPM`, install the library manually by downloading the [Roku SDK][5] zip archive, 
+and unzipping it in your project's root folder.
+
+Make sure you have a `roku_modules/datadogroku` subfolder in both the `components` and ` source` folders of your projet.
```
Comment (cswatt) [inline content/en/real_user_monitoring/roku/_index.md:50]:
```suggestion
Make sure you have a `roku_modules/datadogroku` subfolder in both the `components` and ` source` folders of your project.
```

File: content/en/real_user_monitoring/roku/_index.md [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+
+### Specify application details in the UI
```
Comment (cswatt) [inline content/en/real_user_monitoring/roku/_index.md:52]:
```suggestion
### Specify application details in Datadog
```

File: content/en/real_user_monitoring/roku/_index.md [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+
+1. Navigate to [**UX Monitoring** > **RUM Applications** > **New Application**][2].
+2. Select `Roku` as the application type and enter an application name to generate a unique Datadog application ID and client token.
```
Comment (cswatt) [inline content/en/real_user_monitoring/roku/_index.md:55]:
```suggestion
2. Select **Roku** as the application type and enter an application name to generate a unique Datadog application ID and client token.
```

File: content/en/real_user_monitoring/roku/_index.md [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+3. To disable automatic user data collection for either client IP or geolocation data, uncheck the boxes for those settings. For more information, see [RUM Roku Data Collected][15].
+
+   {{< img src="real_user_monitoring/roku/roku-new-application.png" alt="Create a RUM application for Roku in Datadog" style="width:90%;">}}
+
+To ensure the safety of your data, you must use a client token. If you used only [Datadog API keys][3] to configure the `dd-sdk-roku` library, they would be exposed client-side in the Roku Channel's brightscript code.
```
Comment (cswatt) [inline content/en/real_user_monitoring/roku/_index.md:60]:
```suggestion
To ensure the safety of your data, you must use a client token. If you used only [Datadog API keys][3] to configure the `dd-sdk-roku` library, they would be exposed client-side in the Roku channel's BrightScript code. 
```

File: content/en/real_user_monitoring/roku/_index.md [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+    })
+
+    ' complete your channel setup here
+end sub
+```
+{{< /site-region >}}
+
+### Instrument the channel
+
+See [`Track RUM Resources`][6] to enable automatic tracking of all your views (activities, fragments, and more), and [`Enrich user sessions`][7] to add custom global or user information to your events.
```
Comment (cswatt) [inline content/en/real_user_monitoring/roku/_index.md:154]:
```suggestion
See [**Track RUM Resources**][6] to enable automatic tracking of all your views (activities, fragments, and more), and [**Enrich user sessions**][7] to add custom global or user information to your events.
```

File: content/en/real_user_monitoring/roku/_index.md [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+
+RUM Actions represent the interactions your users have with your channel. You can forward actions to Datadog as follows:
+
+```brightscript
+    targetName = "playButton" ' the name of the SG Node the user interacted with
+    actionType = "click" ' the type of interaction, should be one of "click", "back", or "custom" 
+    m.global.datadogRumAgent.callfunc("addAction", { target: targetName, type: actionType})
+```
+
+#### Track RUM Errors
```
Comment (cswatt) [inline content/en/real_user_monitoring/roku/_index.md:176]:
```suggestion
#### Track RUM errors
```

File: content/en/real_user_monitoring/roku/advanced_configuration.md [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+If you have not set up the SDK yet, follow the [in-app setup instructions][1] or refer to the [Roku RUM setup documentation][2]. 
+
+## Track RUM Resources
+
+### `roUrlTransfer`
+
+Network requests made directly with a `roUrlTransfer` node must be tracked. 
+
+For **synchronous requests**, you can use our wrapper to track the resource automatically, by using our `datadogroku_DdUrlTransfer` wrapper, 
+which supports most features of the `roUrlTransfer` component (except anything related to async network calls).
```
Comment (cswatt) [inline content/en/real_user_monitoring/roku/advanced_configuration.md:33]:
```suggestion
For *synchronous requests*, you can use Datadog's `datadogroku_DdUrlTransfer` wrapper to track the resource automatically. This wrapper supports most features of the `roUrlTransfer` component, but does not support anything related to async network calls.
```

File: content/en/real_user_monitoring/roku/advanced_configuration.md [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+
+```brightscript
+    ddUrlTransfer = datadogroku_DdUrlTransfer(m.global.datadogRumAgent)
+    ddUrlTransfer.SetUrl(url)
+    ddUrlTransfer.EnablePeerVerification(false)
+    ddUrlTransfer.EnableHostVerification(false)
+    result = ddUrlTransfer.GetToString()
+```
+
+For **asynchronous request**, there's no automatic instrumentation yet, meaning you need to track the resource manually. The following code snippet shows how to report the request as a RUM Resource:
```
Comment (cswatt) [inline content/en/real_user_monitoring/roku/advanced_configuration.md:45]:
```suggestion
For *asynchronous requests*, automatic instrumentation is not supported. You need to track the resource manually. The following code snippet shows how to report the request as a RUM Resource:
```

File: content/en/real_user_monitoring/roku/advanced_configuration.md [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+            m.global.datadogRumAgent.callfunc("addResource", msg.getInfo())
+        end if
+    end while
+```
+
+## Enrich user sessions
+
+After your Roku channel is instrumented with RUM, you can further enrich user session information and gain finer control over the attributes collected by tracking custom events.
+
+In addition to the default RUM attributes captured by the RUM Roku SDK automatically, you can choose to add additional contextual information, such as custom attributes, to your RUM events to enrich your observability within Datadog. Custom attributes allow you to slice and dice information about observed user behavior (such as cart value, merchant tier, or ad campaign) with code-level information (such as backend services, session timeline, error logs, and network health).
```
Comment (cswatt) [inline content/en/real_user_monitoring/roku/advanced_configuration.md:108]:
```suggestion
In addition to the default RUM attributes captured by the RUM Roku SDK automatically, you can choose to add additional contextual information, such as custom attributes, to your RUM events to enrich your observability within Datadog. Custom attributes allow you to filter and group information about observed user behavior (such as cart value, merchant tier, or ad campaign) with code-level information (such as backend services, session timeline, error logs, or network health).
```

**Review by cswatt — CHANGES_REQUESTED** [review]

File: content/en/real_user_monitoring/roku/_index.md [hunk 12]
```diff
@@ -0,0 +1,203 @@
...
+{{< partial name="whats-next/whats-next.html" >}}
+
+[1]: https://github.com/DataDog/dd-sdk-android
+[2]: https://app.datadoghq.com/rum/application/create
+[3]: /account_management/api-app-keys/#api-keys
+[4]: /account_management/api-app-keys/#client-tokens
+[5]: https://docs.datadoghq.com/real_user_monitoring/
+[6]: /real_user_monitoring/roku/advanced_configuration/#track-rum-resources
+[7]: /real_user_monitoring/roku/advanced_configuration/#enrich-user-sessions
+[14]: /getting_started/tagging/using_tags/#rum--session-replay
```
Comment (rtrieu) [inline content/en/real_user_monitoring/roku/_index.md:202]:
```suggestion
[8]: /getting_started/tagging/using_tags/#rum--session-replay
```

File: content/en/real_user_monitoring/roku/_index.md [hunk 13]
```diff
@@ -0,0 +1,203 @@
...
+```shell
+ropm install datadog-roku
+```
+
+### Setup manually
+
+If your project does not use `ROPM`, install the library manually by downloading the [Roku SDK][5] zip archive, 
+and unzipping it in your project's root folder.
+
+Make sure you have a `roku_modules/datadogroku` subfolder in both the `components` and ` source` folders of your project.
```
Comment (rtrieu) [inline content/en/real_user_monitoring/roku/_index.md:50]:
```suggestion
Make sure you have a `roku_modules/datadogroku` subfolder in both the `components` and `source` folders of your project.
```

File: content/en/real_user_monitoring/roku/_index.md [hunk 14]
```diff
@@ -0,0 +1,203 @@
...
+
+## Further Reading
+
+{{< partial name="whats-next/whats-next.html" >}}
+
+[1]: https://github.com/DataDog/dd-sdk-android
+[2]: https://app.datadoghq.com/rum/application/create
+[3]: /account_management/api-app-keys/#api-keys
+[4]: /account_management/api-app-keys/#client-tokens
+[5]: https://docs.datadoghq.com/real_user_monitoring/
```
Comment (rtrieu) [inline content/en/real_user_monitoring/roku/_index.md:199]:
```suggestion
[5]: /real_user_monitoring/
```

File: content/en/real_user_monitoring/roku/_index.md [hunk 15]
```diff
@@ -0,0 +1,203 @@
...
+[6]: /real_user_monitoring/roku/advanced_configuration/#track-rum-resources
+[7]: /real_user_monitoring/roku/advanced_configuration/#enrich-user-sessions
+[14]: /getting_started/tagging/using_tags/#rum--session-replay
+[15]: /real_user_monitoring/android/data_collected/
```
Comment (rtrieu) [inline content/en/real_user_monitoring/roku/_index.md:203]:
```suggestion
[9]: /real_user_monitoring/android/data_collected/
```

Comment (rtrieu) [reply content/en/real_user_monitoring/roku/_index.md:203]: Also, I counted 10 links - I think one is unlinked.

File: content/en/real_user_monitoring/roku/_index.md [hunk 16]
```diff
@@ -0,0 +1,203 @@
...
+        env: "<ENV_NAME>",
+        sessionSampleRate: 100, ' the percentage (integer) of sessions to track
+        launchArgs: args
+    })
+
+    ' complete your channel setup here
+end sub
+```
+{{< /site-region >}}
+
```
Comment (rtrieu) [inline content/en/real_user_monitoring/roku/_index.md:151]: {{< site-region region="us,us3,us5,eu,ap1" >}}

File: content/en/real_user_monitoring/roku/_index.md [hunk 17]
```diff
@@ -0,0 +1,203 @@
...
+   {{< img src="real_user_monitoring/roku/roku-new-application.png" alt="Create a RUM application for Roku in Datadog" style="width:90%;">}}
+
+To ensure the safety of your data, you must use a client token. If you used only [Datadog API keys][3] to configure the `dd-sdk-roku` library, they would be exposed client-side in the Roku channel's BrightScript code. 
+
+For more information about setting up a client token, see the [Client Token documentation][4].
+
+### Initialize the library
+
+In the initialization snippet, set an environment name. For more information, see [Using Tags][14].
+
```
Comment (rtrieu) [inline content/en/real_user_monitoring/roku/_index.md:67]: {{< /site-region >}}

File: content/en/real_user_monitoring/roku/_index.md [hunk 18]
```diff
@@ -0,0 +1,203 @@
...
+1. Navigate to [**UX Monitoring** > **RUM Applications** > **New Application**][2].
+2. Select **Roku** as the application type and enter an application name to generate a unique Datadog application ID and client token.
+3. To disable automatic user data collection for either client IP or geolocation data, uncheck the boxes for those settings. For more information, see [RUM Roku Data Collected][15].
+
+   {{< img src="real_user_monitoring/roku/roku-new-application.png" alt="Create a RUM application for Roku in Datadog" style="width:90%;">}}
+
+To ensure the safety of your data, you must use a client token. If you used only [Datadog API keys][3] to configure the `dd-sdk-roku` library, they would be exposed client-side in the Roku channel's BrightScript code. 
+
+For more information about setting up a client token, see the [Client Token documentation][4].
+
```
Comment (rtrieu) [inline content/en/real_user_monitoring/roku/_index.md:63]: {{< /site-region >}}

{{< site-region region="us,us3,us5,eu,ap1" >}}

**Review by rtrieu — CHANGES_REQUESTED** [review]
Hi @xgouchet, as it is, the links are not working on the page because site-regions can't be nested inside of site-regions, so I've added the {{< site-region region="us,us3,us5,eu,ap1" >}} and {{< /site-region}} where appropriate. Can you double check that you've added all the necessary links and the links are correct?

Otherwise, the page looks great. Thanks for adding it!

File: content/en/real_user_monitoring/roku/_index.md (hunk 18 above)
Comment (xgouchet) [inline content/en/real_user_monitoring/roku/_index.md:63]:
```suggestion
{{< /site-region >}}

{{< site-region region="us,us3,us5,eu,ap1" >}}
```

File: content/en/real_user_monitoring/roku/_index.md (hunk 17 above)
Comment (xgouchet) [reply content/en/real_user_monitoring/roku/_index.md:67]:
```suggestion
{{< /site-region >}}
```

File: content/en/real_user_monitoring/roku/_index.md (hunk 16 above)
Comment (xgouchet) [reply content/en/real_user_monitoring/roku/_index.md:151]:
```suggestion
{{< site-region region="us,us3,us5,eu,ap1" >}}
```

File: content/en/real_user_monitoring/roku/_index.md [hunk 19]
```diff
@@ -0,0 +1,203 @@
...
+If your project does not use `ROPM`, install the library manually by downloading the [Roku SDK][5] zip archive, 
+and unzipping it in your project's root folder.
+
+Make sure you have a `roku_modules/datadogroku` subfolder in both the `components` and ` source` folders of your project.
+
+### Specify application details in Datadog
+
+1. Navigate to [**UX Monitoring** > **RUM Applications** > **New Application**][2].
+2. Select **Roku** as the application type and enter an application name to generate a unique Datadog application ID and client token.
+3. To disable automatic user data collection for either client IP or geolocation data, uncheck the boxes for those settings. For more information, see [RUM Roku Data Collected][15].
```
Comment (xgouchet) [inline content/en/real_user_monitoring/roku/_index.md:56]:
```suggestion
3. To disable automatic user data collection for either client IP or geolocation data, uncheck the boxes for those settings. For more information, see [RUM Roku Data Collected][9].
```

File: content/en/real_user_monitoring/roku/_index.md [hunk 20]
```diff
@@ -0,0 +1,203 @@
...
+
+   {{< img src="real_user_monitoring/roku/roku-new-application.png" alt="Create a RUM application for Roku in Datadog" style="width:90%;">}}
+
+To ensure the safety of your data, you must use a client token. If you used only [Datadog API keys][3] to configure the `dd-sdk-roku` library, they would be exposed client-side in the Roku channel's BrightScript code. 
+
+For more information about setting up a client token, see the [Client Token documentation][4].
+
+### Initialize the library
+
+In the initialization snippet, set an environment name. For more information, see [Using Tags][14].
```
Comment (xgouchet) [inline content/en/real_user_monitoring/roku/_index.md:66]:
```suggestion
In the initialization snippet, set an environment name. For more information, see [Using Tags][8].
```

**Review by rtrieu — APPROVED** [review]
This looks great, thanks! @xgouchet is it ready to merge?

File: content/en/real_user_monitoring/roku/data_collected.md [hunk 21]
```diff
@@ -0,0 +1,249 @@
...
+| Long Task  | 15 days   | A long task event is generated for any task in the application that blocks the main thread for more than the specified duration threshold.                                                                                                                                                                                                                                                                                                                                            |
+
+The following diagram illustrates the RUM event hierarchy:
+
+{{< img src="real_user_monitoring/data_collected/event-hierarchy.png" alt="RUM Event hierarchy" style="width:50%;" >}}
+
+## Default attributes
+
+RUM collects common attributes for all events and attributes specific to each event listed below automatically. You can also choose to enrich your user session data by tracking [additional events][2] or by [adding custom attributes][3] to default events specific to your application monitoring and business analytics needs.
```
Comment (timmld) [inline content/en/real_user_monitoring/roku/data_collected.md:46]: Don't know if this is expected at this point, but the links (not only these, but most of the others) are not showing properly in the preview

File: content/en/real_user_monitoring/roku/_index.md [hunk 22]
```diff
@@ -0,0 +1,220 @@
...
+### Initialize the library
+
+In the initialization snippet, set an environment name. For more information, see [Using Tags][8].
+
+[8]: /getting_started/tagging/using_tags/#rum--session-replay
+
+{{< /site-region >}}
+
+{{< site-region region="us" >}}
+```brightscript
```
Comment (timmld) [inline content/en/real_user_monitoring/roku/_index.md:83]: Do we have syntax highlighting for brightscript snippets or not? Looks like not when looking at the preview.

**Review by timmld — CHANGES_REQUESTED** [review]
Thanks @xgouchet for putting this together! There are some things I didn't look at yet, I'll just do another pass later in the day when I find the time

**Review by timmld — APPROVED** [review]

**Pull request merged**